bindings/pyroot/pythonizations/python/ROOT/_pythonization/_rdf_pyz.py

```diff
@@ -0,0 +1,250 @@
+import re
+from .._numbadeclare import _NumbaDeclareDecorator
+
+def _PyFilter(rdf, func_or_str, *args , extra_args = {} ):
+    """
+    Filters the entries of RDF according to a given condition.
+    Arguments:
+        func_or_str: The condition according to which the RDF is to be filtered. 
+            It can be either a python callable or a c style string.
+        *args:
+            Can be at most two of them.
+            a. The list of columns in  which case the starting arguments of the provided function callable in mapped to the columns in the list.
+                If not provide then it tries maps the name of the parameter to a column name of the RDF. Flags an error if not found.
+            b. The name of the Filter as a string.
+        extra_args: If the function needs any other parameters that are not columns and must be passed explicitly.
+    Returns:
+        self: The filtered instance of the RDF.
+    
+    Examples:
+    1. rdf.Filter("x>y")
+        Filter using an c-style expression
+    2. rdf.Filter(lambda x,y: x>y)
+        Filter using a python callable.
+    3. rdf.Filter(lambda x,y: x>y, ["y", "x"])
+        Filter using a python callable with column list passed explicitly.
+        This is equivalent to the filter "y>x".
+    """
+    if isinstance(func_or_str, str): # If string argument is passed. Invoke the Original Filters.
+        return rdf._OriginalFilter(func_or_str, *args)
+
+    if not callable(func_or_str): # The 1st argument is either a string or a python callable.
+        raise ValueError("Filter takes in either a string or a python callable as the first argument.")
+    
+    if len(args) > 2: 
+        raise ValueError("Filter takes in at most 4 arguments (func, cols, filter_name, extra_args).")
```

Comment:
This is a nice error, but we can't write names for the arguments to Filter that are different from the C++ overloads. I suggest changing it to 
```suggestion
    if len(args) > 2: 
        raise TypeError(f"Filter takes at most 3 positional arguments but {len(args) + 1} was given")
```